src/routes/auth/index.ts

```diff
@@ -1,142 +1,288 @@
+			},
+			[StatusCodes.NOT_FOUND]: {
+				description: 'User nof found',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.INTERNAL_SERVER_ERROR]: {
+				description: 'Failed to activate account',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.OKAY]: {
+				description: 'Account activated successfully',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			}
+		}
+	}),
+	async (context) => {
+		const { token } = context.req.valid('params');
+		if (!token) {
+			return context.json({ message: 'Invalid or missing token' }, StatusCodes.BAD_REQUEST);
+		}
+
+		const email = await context.env.ACTIVATION_TOKENS.get(token);
+		if (!email) {
+			return context.json({ message: 'Invalid or expired token' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		const emailExists = await checkEmail(context.env.database, email);
+		if (!emailExists) {
+			return context.json({ message: 'User not found' }, StatusCodes.NOT_FOUND);
+		}
+
+		const activated = await activateProfile(context.env.database, email);
+		if (!activated) {
+			return context.json({ message: 'Failed to activate account' }, StatusCodes.INTERNAL_SERVER_ERROR);
+		}
 
-	const activated = await activateProfile(context.env.database, email);
-	if (!activated) {
-		return context.json({ message: 'Failed to activate account' }, StatusCodes.INTERNAL_SERVER_ERROR);
+		// Remove token after successful activation
+		await context.env.ACTIVATION_TOKENS.delete(token);
+
+		return context.json({ message: 'Account activated successfully' } satisfies StatusResponse, StatusCodes.OKAY);
 	}
+);
 
-	// Remove token after successful activation
-	await context.env.ACTIVATION_TOKENS.delete(token);
+publicAuthRoutes.openapi(
+	createRoute({
+		method: 'post',
+		path: '/',
+		operationId: 'signIn',
+		description: 'Sign in to community hub account',
+		tags: ['Sign-in'],
+		request: {
+			body: { content: { 'application/json': { schema: SignInSchema } }, required: true }
+		},
+		responses: {
+			[StatusCodes.BAD_REQUEST]: {
+				description: 'Invalid JSON format:',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.UNAUTHORIZED]: {
+				description: 'InValid email, profile Id, password or account not activated ',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.NOT_FOUND]: {
+				description: 'Access not found',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.CREATED]: {
+				description: 'Sign in succesful',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			}
+		}
+	}),
+	async (context) => {
+		let parsedBody: SignInData;
 
-	return context.json({ message: 'Account activated successfully' }, StatusCodes.OKAY);
-});
+		try {
+			const body = await context.req.json();
+			parsedBody = SignInSchema.parse(body);
```

Comment:
suggestion: don't parse the request body here let Open API do it based on the schema you declare. The parsed body will be available with something like `const parsedBody = context.req.valid('json')`

Reply:
@abdulqshabbir

Reply:
We can maybe talk a bit tomorrow about it?! But basically all the logic around using a Zod schema to parse the request body.

Reply:
similar to Marco's comment the entire try catch block is not needed

Reply:
Intdocing the suggested code introduces a chain of new errors.

Reply:
Read the hono openapi docs for more examples and details on the specific method to use: https://github.com/honojs/middleware/tree/main/packages/zod-openapi

That should help with the errors